pkg/cephfs/controllerserver.go

```diff
@@ -84,17 +84,25 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 	defer cs.VolumeLocks.Release(requestName)
 
-	volOptions, err := newVolumeOptions(ctx, requestName, req.GetCapacityRange().GetRequiredBytes(),
-		req.GetParameters(), secret)
+	volOptions, err := newVolumeOptions(ctx, requestName, req.GetParameters(), secret)
 	if err != nil {
 		klog.Errorf(util.Log(ctx, "validation and extraction of volume options failed: %v"), err)
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	if req.GetCapacityRange() != nil {
+		volOptions.Size = util.RoundOffBytes(req.GetCapacityRange().GetRequiredBytes())
+	}
+	// TODO need to add check for 0 volume size
```

Comment:
I doubt we get a response at `0` size. It should have been blocked by CO.  have to double confirm though.

Reply:
not sure other CO will not process the vol size of `0`, but as per CSI spec it says volume size must not be negative but it can be zero. even earlier in cephfs, we are able to  create backend images with `0` size